Stripe/STPSourceParams.m

```diff
@@ -202,6 +203,39 @@ + (STPSourceParams *)threeDSecureParamsWithAmount:(NSUInteger)amount
     return params;
 }
 
+#pragma mark - Redirect url
+
+
+/**
+ Private setter allows for setting the name of the app in the returnURL so
+ that it can be displayed on hooks.stripe.com if the automatic redirect back
+ to the app fails.
+ 
+ We intercept the reading of redirect dictionary from STPFormEncoder and replace
+ the value of return_url if necessary
+ */
+- (NSDictionary *)redirectDictionaryWithMerchantNameIfNecessary {
+    if (self.redirectMerchantName
+        && self.redirect[@"return_url"]) {
+        NSURLComponents *urlComponents = [NSURLComponents componentsWithURL:[NSURL URLWithString:self.redirect[@"return_url"]]
+                                                    resolvingAgainstBaseURL:NO];
+        NSMutableArray<NSURLQueryItem *> *queryItems = (urlComponents.queryItems ?: @[]).mutableCopy;
+        [queryItems addObject:[NSURLQueryItem queryItemWithName:@"redirect_merchant_name"
+                                                          value:self.redirectMerchantName]];
+        urlComponents.queryItems = queryItems;
+
+
+        NSMutableDictionary *redirectCopy = self.redirect.mutableCopy;
+        redirectCopy[@"return_url"] = urlComponents.URL;
```

Comment:
No, because everything gets converted to a string using `description` later, but I can manually convert if you want.